ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/HardFork/Combinator/Serialisation/Common.hs

```diff
@@ -371,6 +363,15 @@ data HardForkEncoderException where
   HardForkEncoderFutureEra :: SingleEraInfo blk -> HardForkEncoderException
 
   -- | HFC enabled, but we saw a value from a disabled era
+  --
+  -- This is only thrown by the Node-to-Client encoder. Two nodes' negotiated
+  -- version does not constraint how the /distributed/ chain will evolve, so
+  -- the Node-to-Node communication does not need this. The
+  -- 'Ouroboros.Consensus.Protocol.Praos.Common.MaxMajorProtVer' check will
+  -- enforce it appropriately, and incur explanatory log messages on the node
+  -- that needs to be updated in order to handle the latest hard fork.
```

Comment:
The Haddock link doesn't work here as `MaxMajorProtVer` is only defined in a downstream module (it is rendered as `MaxMajorProtVer` with a non-existing link target). Maybe rephrase to sth like

> If it is desirable to effectively disable certain (eg experimental) eras based on a runtime flag, this needs to be a block-specific mechanism like the @MaxMajorProtVer@-based check in Cardano.

Reply:
> downstream module

Did you mean _downstream package_? I've never realized the package hierarchy mattered (😞), but I'm pretty certain the module hierarchy does not.

Reply:
I'm tempted to leave it as it is. The idea being that even the broken link contains all the info the user would need to find it 🤷 

Moreover, if the Haddock solution improves, then suddenly this markup would suddenly do exactly what we want!

Reply:
> Did you mean *downstream package*? I've never realized the package hierarchy mattered (😞), but I'm pretty certain the module hierarchy does not.

Yes, I meant downstream *package*.

---

Seems fine to leave this as is, was just briefly confused when I locally built the Haddock and the link went nowhere, but everything will be clear as soon as someone looks at the source :+1: